src/flash.rs

```diff
@@ -416,26 +440,29 @@ impl Flash {
         // Ensure no effective write, erase or option byte change operation is ongoing
         while regs.sr.read().bsy().bit_is_set() {}
 
-        // Check and clear (optional) all the error flags due to previous programming/erase operation. Refer to Section 4.7: FLASH error management for details.
-        handle_illegal(regs)?;
+        // 1. Clear all the error flags due to previous programming/erase operation. Refer to Section 4.7: FLASH error management for details.
+        clear_error_flags(regs);
 
-        // 1. Unlock the FLASH_CR1/2 register, as described in Section 4.5.1: FLASH configuration protection (only if register is not already unlocked).
+        // 2. Unlock the FLASH_CR1/2 register, as described in Section 4.5.1: FLASH configuration protection (only if register is not already unlocked).
         self.unlock(bank)?;
 
-        // 2. Enable write operations by setting the PG1/2 bit in the FLASH_CR1/2 register.
-        regs.cr.modify(|_, w| w.pg().set_bit().fw().set_bit());
+        // 3. Enable double-word parallelism.
+        unsafe { regs.cr.modify(|_, w| w.psize().bits(0b11)) }
 
-        // 3. Check the protection of the targeted memory area.
+        // 4. Enable write operations by setting the PG1/2 bit in the FLASH_CR1/2 register.
+        regs.cr.modify(|_, w| w.pg().set_bit());
+
+        // 5. Check the protection of the targeted memory area.
         // SKIP: In standard mode the entire UserBank1 and UserBank2 should have no write protections
 
-        // 4. Write one Flash-word corresponding to 32-byte data starting at a 32-byte aligned address.
         let mut addr = bank.sector_address(sector) as *mut u32;
 
         // Offset it by the start position
         addr = unsafe { addr.add(start / 4) };
 
-        // Iterate on chunks of 32bit
-        for chunk in data.chunks_exact(4) {
+        // 6. Write a double-word corresponding to 32-byte data starting at a 32-byte aligned address.
+        // TODO: Use `.array_chunks` when it's stable.
```

Comment:
What does `array_chunks` give us over just `chunks_exact`? Seems like the existing impl is fine and we can remove the TODO?

Reply:
It would be a little cleaner (wouldn't need a `try_into` lower down), but I'm pretty sure it'll get optimized out, so doesn't really matter!~

Reply:
Removed!